sdk/core/core-lro/docs/LROEngine.md

```diff
@@ -0,0 +1,213 @@
+    flatResponse: unknown
+  ) => boolean
+): Promise<LroResponse<T>> {
+  const response = await this.sendOperation(this.args, this.spec); // the class will have sendOperation, args, and spec as private fields
+  initializeState(response.rawResponse, response.flatResponse);
+  return response;
+}
+```
+
+#### `sendPollRequest`
+
+This method should be implemented to send a polling request, a request the service should respond to with the status of the operation, and it has the following signature:
+
+```ts
+sendPollRequest: (config: LroConfig, path: string) => Promise<LroStatus<T>>
+```
+
+This method takes two things as input, it takes the request path as you would expect but also takes a parameter of type `LroConfig`. The latter is an object that holds meta information about how the Lro should work and clients could need access to this information in order to do things like doing custom de-serialization in certain cases where the swagger does not describe all response shapes. The `LroConfig` interface is defined as follows:
+
+```ts
+export interface LroConfig {
+  mode?: LroMode;
+  resourceLocation?: string;
+}
+```
+
+The information in it is currently populated from the response of the initial request. `mode` specifies which LRO implementation this particular API follows (see the [Terminology](#terminology) section). Furthermore, `resourceLocation` is a path where the desired result of the LRO could potentially be located.
+
+Implementing `sendPollRequest` is tricky because it returns a `LroStatus` which has a `done` field and it is not easy to figure out whether the LRO has finished, so this design exposes a helper function, called `createGetLroStatusFromResponse`, which the implementation should call to get a `getLroStatusFromResponse`, which in turn should be called on the response to detect whether the operation has finished at this point. Here is what the implementation looks like for `@azure/core-http`, ignoring the custom de-serialization stuff:
+
+```ts
+public async sendPollRequest(
+    config: LroConfig,
+    path: string
+): Promise<LroStatus<T>> {
+  const getLroStatusFromResponse = createGetLroStatusFromResponse(this, config, this.finalStateVia);
+  const { flatResponse, rawResponse } = await sendOperation(this.args, {
+    ...this.spec,
+    httpMethod: "GET",
+    path: path)
+  });
+  return getLroStatusFromResponse(rawResponse, flatResponse as T);
+}
+```
+
+#### `retrieveAzureAsyncResource`
+
+This method is only needed to potentially retrieve the provisioned azure resource in the _Azure Async Operation_ mode after the operation has finished and it should be different from the `sendPollRequest` one in two things:
+
+- should set the obscure `shouldDeserialize` option needed for doing custom de-serialization on the result if it is supported by the client
+- we already know the operation has finished at this point, so `done` can be set to `true` right away without calling `getLroStatusFromResponse`
```

Comment:
I think we could it make it easier for clients to implement by setting done to true internally, and change the return type to be just `LroResponse` instead of `LroStatus`.